OpenRA.Mods.Common/AI/SupportPowerDecision.cs

```diff
@@ -94,7 +94,7 @@ public int GetAttractiveness(IEnumerable<Actor> actors, Player firedBy)
 		/// <summary>Makes up part of a decision, describing how to evaluate a target.</summary>
 		public class Consideration
 		{
-			public enum DecisionMetric { Health, Value, None }
+			public enum DecisionMetric { Health, Value, BaseBuilding, None }
```

Comment:
My example has `Attractiveness: 10` on ConYards. The `Value` metric doesn't really matter, because even if you use `Health`, the code will acknowledge that the conyard satisfies the `ConYard` `TargetType` then and will acknowledge the conyard-specific weight increase (`None` would be the really tough override though, see below).

The design is really simple. If it has `None`, it's ignoring the properties of the actor which satisfies the properties specified and just returns the maximum attractiveness - in our case that the actor has the `ConYard` `TargetType` specified. If it's `Health`, then it will try to look at the satisfying actor's health as well and only accumulate the maximum attractiveness if the satisfying actor has full health. If it's `Value` then this attractiveness is instead based on the satisfying actor's worth - which doesn't make much sense on conyards per se, since all mods have the same properties on conyards, but if we step back and set `Types` to `Structure` then a conyard will still have more attractiveness than a silo because said conyard has 2000+ worth in all mods, while a silo has 1000- afterall.

So if you use a quite big base `Attractiveness` value, and a really exact kind of `(Target)Types`, then the `TargetMetric` practically doesn't even matter, be it `None`, `Value` or `Health`, because even the fragment attractiveness you end up with will be enough to prioritize the targets you're looking for.

Documentation is in the code and outputted to https://github.com/OpenRA/OpenRA/wiki/Traits-%28playtest%29 via the descs. Feel free to read.

Also feel free to actually comprehend a switch-case you're actually editing, because what you did is equivalent of the `None` `TargetMetric` on actors which have the `BaseBuilding` trait, which again can just be substituted of giving a specific `TargetType` to these actors and just using `Types: specific TargetType` with `TargetMetric` `None`.